change spinlock to be raw spinlock when serializing prints from NMI

From: Clark Williams <williams@redhat.com>


---

 arch/x86/kernel/nmi_32.c |    2 +-
 arch/x86/kernel/nmi_64.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.24.7-rt27/arch/x86/kernel/nmi_32.c
===================================================================
--- linux-2.6.24.7-rt27.orig/arch/x86/kernel/nmi_32.c	2009-02-08 00:04:41.000000000 -0500
+++ linux-2.6.24.7-rt27/arch/x86/kernel/nmi_32.c	2009-02-08 00:05:02.000000000 -0500
@@ -401,7 +401,7 @@ nmi_watchdog_tick(struct pt_regs * regs,
 	}
 
 	if (cpu_isset(cpu, backtrace_mask)) {
-		static DEFINE_SPINLOCK(lock);	/* Serialise the printks */
+		static DEFINE_RAW_SPINLOCK(lock);	/* Serialise the printks */
 
 		spin_lock(&lock);
 		printk("NMI backtrace for cpu %d\n", cpu);
Index: linux-2.6.24.7-rt27/arch/x86/kernel/nmi_64.c
===================================================================
--- linux-2.6.24.7-rt27.orig/arch/x86/kernel/nmi_64.c	2009-02-08 00:04:41.000000000 -0500
+++ linux-2.6.24.7-rt27/arch/x86/kernel/nmi_64.c	2009-02-08 00:05:02.000000000 -0500
@@ -393,7 +393,7 @@ nmi_watchdog_tick(struct pt_regs * regs,
 	}
 
 	if (cpu_isset(cpu, backtrace_mask)) {
-		static DEFINE_SPINLOCK(lock);	/* Serialise the printks */
+		static DEFINE_RAW_SPINLOCK(lock);	/* Serialise the printks */
 
 		spin_lock(&lock);
 		printk("NMI backtrace for cpu %d\n", cpu);
